### ISSUE TITLE
stop cronjob run during beforeRun event

### ISSUE THREAD

**Issue by larsbo** [liebig/cron#83]
Using the `cron.beforeRun` event I want to implement something like this (which should stop the job running on a holiday):
```php
\Event::listen('cron.beforeRun', function($rundate) {
    $date = date('Y-m-d', $rundate);

    /** @var Holidays $holidays */
    $holidays = new Holidays();

    if ($holidays->isHoliday($date)) {
        return false;
    }
});
```
I think the current `cron.beforeRun` implementation does not allow preventing the run but this would be a nice feature, wouldn't it?

**Comment by liebig** [comment]
Yes, this would be a nice feature. But you can use the `cron.collectJobs` event for your purpose:
```php
\Event::listen('cron.collectJobs', function() {
    $date = date('Y-m-d', $rundate);
    /** @var Holidays $holidays */
    $holidays = new Holidays();

    if ($holidays->isHoliday($date)) {
        \Cron::add('HolidayJob1', '* * * * *', function() {
            // Do some crazy things unsuccessfully every minute
            return 'No';
        });
        \Cron::add('HolidayJob2', '* * * * *', function() {
            // Do some crazy things unsuccessfully every minute
            return 'No';
        });
    } else {
        \Cron::add('RegularJob1', '* * * * *', function() {
            // Do some crazy things unsuccessfully every minute
            return 'No';
        });
        \Cron::add('RegularJob2', '* * * * *', function() {
            // Do some crazy things unsuccessfully every minute
            return 'No';
        });
    }
```
To my mind, this would be a cleaner way, because Cron is still running (and the in time check would still work) and if you need jobs during holidays, you can add them. Maybe this is a better way instead of disabling Cron completely. What do you think? Will this solution work for you?

**Comment by larsbo** [comment]
well, that makes a lot of sense 😃 

I misinterpreted the event as fired just before every single cronjob. Looking into the code I found out it fires at the beginning of a complete run. So this can't work if you want to distinguish between jobs that should run and jobs that shouldn't (as you pointed out in your answer).

Therefore using the `cron.collectJobs` event is indeed a better and cleaner position for making addional checks.

Thank you for making this clear 👍
